index.js

```diff
@@ -0,0 +1,141 @@
+"use strict";
+
+module.exports = function preset(context, opts = []) {
+  // see https://github.com/facebook/create-react-app/blob/590df7eead1a2526828aa36ceff41397e82bd4dd/packages/babel-preset-react-app/index.js#L52
+  const env = process.env.BABEL_ENV || process.env.NODE_ENV;
+  if (
+    env !== undefined &&
+    env !== "development" &&
+    env !== "test" &&
+    env !== "production"
+  ) {
+    throw new Error(
+      "Using `@mapbox/babel-preset-mapbox` requires that you specify `NODE_ENV` or " +
+        '`BABEL_ENV` environment variables. Valid values are "development", ' +
+        '"test", and "production". Instead, received: ' +
+        JSON.stringify(env) +
+        "."
+    );
+  }
+
+  if (!Array.isArray(opts)) {
+    throw new Error(
+      "`@mapbox/babel-preset-mapbox` expects an array as an option."
+    );
+  }
+
+  // a shorthand if you want to override one property
+  if (opts.length === 2 && typeof opts[0] === "string") {
+    opts = [opts];
+  }
+
+  const isEnvDevelopment = env === "development" || env === undefined;
+  const isEnvProduction = env === "production";
+  const isEnvTest = env === "test";
+  let payload = [
+    { name: "babel-preset-react" },
+    {
+      // Necessary to include regardless of the environment because
+      // in practice some other transforms (such as object-rest-spread)
+      // don't work without it: https://github.com/babel/babel/issues/7215
+      name: "babel-plugin-transform-es2015-destructuring"
+    },
+    {
+      name: "babel-plugin-transform-class-properties"
+    },
+    {
+      // useBuiltIns for these plugins means they'll rely on Object.assign
+      // being available, instead of adding their own polyfill.
+      name: "babel-plugin-transform-object-rest-spread",
+      options: { useBuiltIns: true }
+    },
+    {
+      name: "babel-plugin-transform-react-jsx",
+      options: { useBuiltIns: true }
+    },
+    {
+      name: "babel-plugin-transform-runtime",
+      options: {
+        helpers: false,
+        polyfill: false,
+        regenerator: true
+      }
+    }
```

Comment:
Ah, so decided to allow for async/await despite debugging troubles?

Reply:
Ok. We can add a caveat that you should know what you're getting into if you use async/await.